dagster_polars/io_managers/parquet.py

```diff
@@ -74,30 +51,26 @@ def scan_parquet(path: "UPath", context: InputContext) -> pl.LazyFrame:
     :param context:
     :return:
     """
-    assert context.metadata is not None
-
-    storage_options: Optional[dict[str, Any]] = None
+    context_metadata = context.metadata or {}
 
-    try:
-        storage_options = path.storage_options
-    except AttributeError:
-        # TODO: explore removing this as universal-pathlib should always provide storage_options in newer versions
-        pass
+    storage_options: Optional[dict[str, Any]] = path.storage_options if hasattr(path, "storage_options") else None
```

Comment:
shouldn't `self.cloud_storage_options` be used here?

Reply:
No, we are calling here the property `storage_options` which is inherited from the UPathIOmanager and actually grabs the storage_options from the UPath object _base_path.